TestFoundation/TestDateFormatter.swift

```diff
@@ -432,15 +432,45 @@ class TestDateFormatter: XCTestCase {
         
         formatter.locale = Locale(identifier: "ja_JP")
         formatter.calendar = Calendar(identifier: .japanese)
-        formatter.dateFormat = "Gy年M月dd日 HH:mm"
+        formatter.dateFormat = "Gy年M月d日 HH:mm"
         formatter.timeZone = TimeZone(abbreviation: "JST")
         
-        // parse test
-        let parsed = formatter.date(from: "平成31年4月30日 23:10")
-        XCTAssertEqual(parsed?.timeIntervalSince1970, 1556633400) // April 30, 2019, 11:10 PM (JST)
+        do {
+            // parse test
+            let parsed = formatter.date(from: "平成31年4月30日 23:10")
+            XCTAssertEqual(parsed?.timeIntervalSince1970, 1556633400) // April 30, 2019, 11:10 PM (JST)
+            
+            // format test
+            let dateString = formatter.string(from: Date(timeIntervalSince1970: 1556633400)) // April 30, 2019, 11:10 PM (JST)
+            XCTAssertEqual(dateString, "平成31年4月30日 23:10")
+        }
+        
+        // Test for new Japanese era (starting from May 1, 2019)
+        do {
+            // parse test
+            let parsed = formatter.date(from: "令和1年5月1日 23:10")
+            XCTAssertEqual(parsed?.timeIntervalSince1970, 1556719800) // May 1st, 2019, 11:10 PM (JST)
+            
+            // format test
+            let dateString = formatter.string(from: Date(timeIntervalSince1970: 1556719800)) // May 1st, 2019, 11:10 PM (JST)
+            XCTAssertEqual(dateString, "令和1年5月1日 23:10")
+        }
+        
+        // Test for new Japanese era and 元年 representaion of 1st year
+        // This block needs ICU 64.2
         
-        // format test
-        let dateString = formatter.string(from: Date(timeIntervalSince1970: 1556633400)) // April 30, 2019, 11:10 PM (JST)
-        XCTAssertEqual(dateString, "平成31年4月30日 23:10")
+// Disabled until the ICU is updated.
```

Comment:
Could you enable these tests please. I just want to check they fail with the current ICU so then they pass with the new ICU we know its all working correctly, thanks.

Reply:
@spevans This following codes require ICU 64.2. It should be enabled?
And if the ICU is 64.2, L448-L457 will fail. This lines for ICU 61.2.

Reply:
We cannot commit tests that fail.

If you want, you can replace the `do` block with `if shouldAttemptXFailTests("These tests require ICU 64.2 or later. https://bugs.swift.org/browse/SR-XXXX") { … }` where the XXXX link is a bug filed on bugs.swift.org to migrate to newer ICU. @spevans, you can then run this test by setting the environment variable `NS_FOUNDATION_ATTEMPT_XFAIL_TESTS` to `YES`.